tests/upgrade-tests/test_upgrade_snaps.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+from utils import temporary_model, wait_for_ready, assert_healthy
+
+test_cases = [
+    # bundle                 from_channel  to_channel
+    ('kubernetes-core',      '1.6/stable', '1.6/edge'),
+    ('kubernetes-core',      '1.6/stable', '1.7/edge'),
+    ('canonical-kubernetes', '1.6/stable', '1.6/edge'),
+    ('canonical-kubernetes', '1.6/stable', '1.7/edge'),
+]
+
+
+async def set_snap_channel(model, channel):
+    master = model.applications['kubernetes-master']
+    await master.set_config({'channel': channel})
+    worker = model.applications['kubernetes-worker']
+    await worker.set_config({'channel': channel})
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('bundle,from_channel,to_channel', test_cases)
+async def test_upgrade_snaps(bundle, from_channel, to_channel):
+    async with temporary_model() as model:
+        await model.deploy(bundle, channel='stable')
```

Comment:
This should probably be `channel='edge'`. I'm still thinking about it but if anyone has any opinions, lemme know!